### DISCUSSION TITLE
TypeScript support

### DISCUSSION THREAD

**Discussion by GerardSmit** [nandorojo/dripsy#138] [Q&A] [answered]
After following https://www.dripsy.xyz/typescript/setup, I've had the issue that autocomplete wasn't working.
It turns out that the typings couldn't be loaded:

![image](https://user-images.githubusercontent.com/2109929/139487250-772ca07c-01b8-4f9f-b181-f5adf0e5c7cd.png)

After some digging, it turns out that `Function.Narrow` doesn't exists:

![image](https://user-images.githubusercontent.com/2109929/139487348-53eb7990-91c9-46a5-9d56-7521647c18a7.png)

This is caused by Visual Studio Code loading `ts-toolbelt` from the TypeScript installation (`AppData/Local/Microsoft/TypeScript/4.4/node_modules/ts-toolbelt`). This version doesn't have `Function.Narrow`.

You can disable this in VS Code by adding the following in your `settings.json`:
```json
"typescript.disableAutomaticTypeAcquisition": true
```

Sadly enough, it didn't work directly since `ts-toolbelt` is a dependency of another package:
```
$ npm list ts-toolbelt
+-- dripsy@3.1.0
| `-- @dripsy/core@3.1.0
|   `-- ts-toolbelt@9.6.0 deduped
+-- react-native-render-html@6.2.0
| `-- @types/ramda@0.27.46
|   `-- ts-toolbelt@6.15.5
`-- ts-toolbelt@9.6.0
```

I don't have a solution for this yet, other than removing this version of `ts-toolbelt` from `node_modules\@types\ramda\node_modules`.

I have autocompletion working but removing a folder in `node_modules` feels a little bit hacky. Does anyone know how to resolve this?

To reproduce this problem, simply install `react-native-render-html` (or `@types/ramda@0.27.46`).

**Comment by nandorojo** [comment, marked as answer]
Hmm you’re using makeTheme right?

what if you use yarn resolutions for ts-toolbelt?

**Reply by nandorojo** [reply]
`package.json`

```json
{
  "resolutions": {
    "ts-toolbelt": "9.6.0"
  }
}
```

Then run `yarn`

**Reply by GerardSmit** [reply]
It's working perfectly. 

I didn't know Yarn resolutions was a thing. I was using NPM for this project but this is worth the switch.

Thanks for the solution 😄

**Reply by nandorojo** [reply]
Awesome!
